clients/client/src/main/java/org/projectnessie/client/auth/BasicAuthenticationProvider.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.client.auth;
+
+import static org.projectnessie.client.NessieConfigConstants.CONF_NESSIE_PASSWORD;
+import static org.projectnessie.client.NessieConfigConstants.CONF_NESSIE_USERNAME;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.function.Function;
+import org.projectnessie.client.http.HttpAuthentication;
+import org.projectnessie.client.http.RequestContext;
+import org.projectnessie.client.http.RequestFilter;
+
+/**
+ * HTTP BASIC authentication provider.
+ *
+ * <p>Takes parameters {@link org.projectnessie.client.NessieConfigConstants#CONF_NESSIE_USERNAME}
+ * and {@link org.projectnessie.client.NessieConfigConstants#CONF_NESSIE_PASSWORD}.
+ */
+public class BasicAuthenticationProvider implements NessieAuthenticationProvider {
+
+  public static final String AUTH_TYPE_VALUE = "BASIC";
+
+  public static NessieAuthentication of(String username, String password) {
```

Comment:
Could we return `HttpAuthentication` to be more specific?

Reply:
nit: `BasicAuthenticationProvider.of(name,pwd)` looks odd to me because it return an "authentication" not a provider...

How about:
```
public class BasicHttpAuthentication implements NessieAuthenticationProvider {
  public static HttpAuthentication of(String username, String password) {... }
}
}

Reply:
Hm - I wanted to avoid exposing "internal" classes. Maybe rename `of` to `create`?

Reply:
`create` SGTM :+1: